src/plugins/adapters/AnalyticsTrackerAdapter.js

```diff
@@ -77,6 +77,23 @@ export class AnalyticsTrackerAdapter {
     return this
   }
 
+  /**
+   * Adds a user-signed-up event to the events array.
+   *
+   * @param {Object} payload - The payload containing the sign-up details.
+   * @param {string} payload.method - The method of sign-up, such as (email | gitHub | google).
```

Comment:
para melhorar o intellisense, usar @param {'email' | 'gitHub'| 'google'} payload.method - The method of sign-up

Reply:
dessa forma não será indicado ao dev usar qualquer valor de string ao executar o método, mas apenas essas strings.

Reply:
Atualizada a JSdoc